python/core/working-with-strings/efficient-concatenation-with-join.md

```diff
@@ -100,3 +105,20 @@ l = "".join(l)
 * Enki
 * E n k i
 * E,n,k,i
+
+---
+## Footnotes
+
+[1:Efficient Concatenation]
+To understand why iteratively creating a string with `+=` is inefficient, let's take a look at this example:
+```py
+x = 'en'
+x += 'ki' # 'enki' 
+x += 'enki' # 'enkienki'
+```
+
+In this example, Python first allocates and creates `'enki'`, before it can allocate and create `'enkienki'`. When concatenating a small number of strings this wouldn't pose a problem, but what would happen if the total number of string that need to be concatenated was over 1000? In this case, each substring would get copied approximately `N/2` times (where `N` represents the number of strings to be concatenated). You can already see why this method does not scale well with a large number of strings.
```

Comment:
How about?

```suggestion
In this example, Python first allocates the memory for `'en'`, then again for `en` and `ki` (to make `enki`), then again for `en` and `ki` and `enki` (to make `'enkienki'`). This means that each string from the previous concatenation recreates more memory for the next concatenation even though it already had enough memory to fit in. In this scenario, instead of just creating the memory for `enkienki`, we would end up creating that memory plus the memory needed for `en` and `enki` during prior concatenations. 
When concatenating a small number of strings this wouldn't pose much of a problem, but what would happen if the total number of strings that need to be concatenated was 1000? Since each substring gets re-created for all concatenations that follow it, the first substring would get created 1000 times, the second one 999 times, the third one 998 times, and so forth, which averages to about ~500 concatenations per substring. This means that concatenation of 1000 strings  will end up with approximately 500,000 concatenations! In general terms, concatenating `N` strings would lead approximately to `(N*N)/2` concatenations, which is extremely inefficient. This is why we must handle Python's string immutability with caution.
```